valgrind/docker/Dockerfile

```diff
@@ -0,0 +1,37 @@
+FROM ubuntu:24.04
+
+RUN apt-get update && apt-get install -y git
+RUN git clone https://github.com/theory/pgenv
+
+WORKDIR /pgenv
+RUN git fetch && git checkout v1.3.6b
+
+RUN apt-get update && apt-get install -y make curl
+RUN ./bin/pgenv config init
+RUN sed -i 's/^PGENV_CONFIGURE_OPTIONS=.*/PGENV_CONFIGURE_OPTIONS=([0]="--enable-debug" [1]="--enable-depend" [2]="--enable-cassert" [3]="CFLAGS=-ggdb -Og -fno-omit-frame-pointer -DUSE_VALGRIND")/' ./config/default.conf
+RUN sed -i "s/^PGENV_MAKE_OPTIONS=.*/PGENV_MAKE_OPTIONS=([0]='-sj$(nproc)')/" ./config/default.conf
+
+RUN apt-get update && apt-get install -y bzip2 build-essential libicu-dev pkg-config flex bison liblz4-dev libreadline-dev libzstd-dev libxslt1-dev zlib1g-dev zlib1g-dev libxml2-utils docbook-xsl xsltproc valgrind
+ARG PG_VERSION
+RUN ./bin/pgenv build $PG_VERSION && ./bin/pgenv switch $PG_VERSION
```

Comment:
you will want to add some makeflags here to speed up the build to use all available cores.

https://github.com/citusdata/citus/blob/117bd1d04fe55b4001012b3923216d5df0fda10e/.devcontainer/Dockerfile#L71